### ISSUE TITLE
not recognize "*, " in python

### ISSUE THREAD

**Issue by zhangzq** [doxygen/doxygen#10838]
**Describe the bug**
The python code:

```python
def fun1(one, *two):
    pass

def fun2(one, *, two):
    pass
```

will output the same functions:

```python
fun1 (one, *two)
fun2 (one, *two)
```

But actuall the `two` in these two functions are not the same. ref <https://peps.python.org/pep-3102/>.

 
**To Reproduce**
The above codes and default doxyfile.

**Expected behavior**
The fun2 should be :

```
fun2(one, *, two)
```

**Version** 
1.11.0 (74f8b41aead51679bf364056b4fd091cf828a7cf)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10841

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
